docs/framework/guides/tutorials/data-from-external-source.md

```diff
@@ -0,0 +1,877 @@
+		plugins: [ Essentials, Paragraph, Heading, List, Bold, Italic, ExternalWidget ],
+		toolbar: [ 'heading', 'bold', 'italic', 'numberedList', 'bulletedList', '|', 'undo', 'redo' ]
+	} )
+	.then( editor => {
+		console.log( 'Editor was initialized', editor );
+
+		CKEditorInspector.attach( 'editor', editor );
+
+		// Expose for playing in the console.
+		window.editor = editor;
+	} )
+	.catch( error => {
+		console.error( error.stack );
+	} );
+```
+
+
+Before building the project you still need to define the `ExternalWidget` plugin. The project will have a structure as below:
+
+```
+├── app.js
+├── dist
+│   ├── bundle.js
+│   └── bundle.js.map
+├── index.html
+├── node_modules
+├── package.json
+├── external-widget
+│   ├── externalwidget.js
+│   ├── externalwidgetcommand.js
+│   ├── externalwidgetediting.js
+│   ├── externalwidgetui.js
+│   └── theme
+│       └── externalwidget.css
+│
+│   ... the rest of the plugin files go here as well.
+│
+└── webpack.config.js
+```
+
+
+You can see that the external widget feature has an established plugin structure: the master (glue) plugin (`external-widget/externalwidget.js`), the "editing" (`external-widget/externalwidgetediting.js`) and the "UI" (`external-widget/externalwidgetui.js`) parts.
+
+The master (glue) plugin:
+
+```js
+// external-widget/externalwidget.js
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+import ExternalWidgetEditing from './externalwidgetediting';
+import ExternalWidgetUI from './externalwidgetui';
+
+export default class ExternalWidget extends Plugin {
+	static get requires() {
+		return [ ExternalWidgetEditing, ExternalWidgetUI ];
+	}
+}
+```
+
+The UI part (empty for now):
+
+```js
+// external-widget/externalwidgetui.js
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+export default class ExternalWidgetUI extends Plugin {
+	init() {
+		console.log( 'ExternalWidgetUI#init() got called' );
+	}
+}
+```
+
+And the editing part (empty for now):
+
+```js
+// external-widget/externalwidgetediting.js
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+export default class ExternalWidgetEditing extends Plugin {
+	init() {
+		console.log( 'ExternalWidgetEditing#init() got called' );
+	}
+}
+```
+
+At this stage you can build the project and open it in the browser to verify if it is building correctly.
+
+Use this command `./node_modules/.bin/webpack --mode development` in root folder of the widget to build the project.
+
+After the build is completed, open `index.html` in your browser to check if all is correct at this stage.
+
+## The model and the view layers
+
+The external widget feature will be {@link module:engine/model/schema~SchemaItemDefinition defined as an inline} (text-like) element so it will be inserted into other editor blocks, like `<paragraph>`, that allow text. The external widget will have a `data-resource-url` attribute. This means that the model containing some text and a external widget will look like this:
```

Comment:
```suggestion
The external widget feature will be {@link module:engine/model/schema~SchemaItemDefinition defined as an inline} (text-like) element so it will be inserted into other editor blocks that allow text e.g. `<paragraph>`. The external widget will have a `data-resource-url` attribute. This means that the model containing some text and an external widget will look like this:
```